src/components/updatemenus/draw.js

```diff
@@ -315,8 +337,32 @@ function drawButtons(gd, gHeader, gButton, menuOpts) {
 
     buttons.call(styleButtons, menuOpts);
 
-    // translate button group
-    Lib.setTranslate(gButton, menuOpts.lx, menuOpts.ly);
+    scrollBox.enable();
+
+    var active = menuOpts.active,
+        i;
+    if(isVertical) {
+        if(scrollBox._vbar) {
+            var translateY = 0;
+            for(i = 0; i < active; i++) {
+                translateY += menuOpts.heights[i] + constants.gapButton;
```

Comment:
This looks off for `direction: 'up'` when the updatemenu `y` isn't the default value:

```js
var direction = 'up',
    data = [],
    layout = {
        updatemenus: [{
            y: 0,
            direction: direction,
            buttons: []
        }]
    };

for (var i = 0, n = 20; i < n; i++) {
    data.push({
        y: Array.apply(null, Array(10)).map(() => Math.random()),
        line: {
            shape: 'spline',
            color: 'red'
        },
        visible: i === 0,
        name: 'Data set ' + i,
    });

    var visible = Array.apply(null, Array(n)).map((_, j) => i === j);

    layout.updatemenus[0].buttons.push({
        method: 'restyle',
        args: ['visible', visible],
        label: 'Data set ' + i
    });
}

Plotly.plot('graph', data, layout);
```

gives (after clicking on the header)

![image](https://cloud.githubusercontent.com/assets/6675409/22479015/30b7cd10-e7ba-11e6-9e42-a213746bf283.png)